### PR TITLE
[FIX] stock : Any change in settings triggered a change of show_opera…

### PR DESCRIPTION
**Pull request by nmarotte**
…tions

Issue: When changing a settings and saving, all the operation types (stock_picking.type) that had a code different than incoming had their Show Detailed Operations (show_operations) forced to be True, even when the change in settings was totally unrelated

Steps to reproduce :
 1) Go to Inventory (stock) / Configuration / Warehouse Management / Operations Types
 2) Find or create an Operation Type with Type of Operation set to something else than Receipt (incoming)
 3) Ensures that the checkbox Show Detailed Operations (show_operations) is **unchecked**
 4) Go to Settings and check or uncheck any setting, for example Permissions > Default Access Rights
 5) Save
 6) Go back to the Operation Type seen before
 7) Show Detailed Operation is now **checked**

Why is that a bug:
 The intended behaviour was to check that Show Detailed Operations box every time we change group_stock_multi_locations, group_stock_production_lot, or group_stock_tracking_lot from **unchecked to checked**, but the actual behaviour was to check that SDO box every time one of the **3 was checked when changing settings**

opw-2604730

### CODE REVIEW

File: addons/stock/models/res_config_settings.py [hunk 1]
```diff
@@ -69,6 +69,7 @@ def set_values(self):
         if not self.group_stock_multi_locations and location_grp in base_user.implied_ids and warehouse_grp in base_user.implied_ids:
             raise UserError(_("You can't desactivate the multi-location if you have more than once warehouse by company"))
 
+        previous = self.default_get([])
```
Comment (Feyensv) [inline addons/stock/models/res_config_settings.py:72]:
```suggestion
        previous = self.default_get(['group_stock_multi_locations', 'group_stock_production_lot', 'group_stock_tracking_lot'])
```
Please at least restrict the `default_get` call to the necessary fields.

Comment (nmarotte) [reply addons/stock/models/res_config_settings.py]: I agreed, I updated the commit 😄

**Comment by nmarotte** [comment]
@odoo/rd-logistics Hey, I think I still need a review from you guys, could you do that when you have the time ? Thanks 😄

**Comment by ryv-odoo** [comment]
Hello @nmarotte ,
It seems correct to me but there is the same kind of error for this block of code (just above): 
```python
        warehouse_obj = self.env['stock.warehouse']
        if self.group_stock_multi_locations:
            # override active_test that is false in set_values
            warehouses = warehouse_obj.with_context(active_test=True).search([])
            active = True
        else:
            warehouses = warehouse_obj.search([
                ('reception_steps', '=', 'one_step'),
                ('delivery_steps', '=', 'ship_only')])
            active = False
        warehouses.mapped('int_type_id').write({'active': active})
```
Maybe we should fix it too (use your `previous` variable - Maybe you shoul name it `previous_group`).

**Comment by nmarotte** [comment]
Hey @ryv-odoo, thanks for the feedback, I've applied the same reasoning for the code block above, I hope it's all good for you :smile:
Don't hesitate to comment (or PM on discord, look for (nama)) if there is anything else

Best,
Nathan

File: addons/stock/models/res_config_settings.py [hunk 2]
```diff
@@ -69,28 +69,29 @@ def set_values(self):
...
+            warehouse_obj.search([
                 ('reception_steps', '=', 'one_step'),
-                ('delivery_steps', '=', 'ship_only')])
-            active = False
-        warehouses.mapped('int_type_id').write({'active': active})
+                ('delivery_steps', '=', 'ship_only')]
+            ).mapped('int_type_id').write({'active': False})
 
-        if self.group_stock_multi_locations or self.group_stock_production_lot or self.group_stock_tracking_lot:
+        if (self.group_stock_multi_locations and not previous_group.get('group_stock_multi_locations')) or\
```
Comment (ryv-odoo) [inline addons/stock/models/res_config_settings.py:92]: Suggestion to simplify and avoid multiple line condition
```python
if any(self[group] and not prev_value for group, prev_value in previous_group.items()):
```

**Comment by ryv-odoo** [comment]
LGTM, thank you @nmarotte 
@robodoo r+

**Pull request closed without merging**